packages/services/src/services/Locale/Locale.js

```diff
@@ -0,0 +1,143 @@
+import root from 'window-or-global';
+import axios from 'axios';
+import { ipcinfoCookie, geolocation } from '@carbon/ibmdotcom-utilities';
+
+/**
+ * @constant {string | string} Host for the Locale API call
+ * @private
+ */
+const _host = process.env.TRANSLATION_HOST || 'https://www.ibm.com';
+
+/**
+ * @constant {string | string} CORS proxy for lower environment calls
+ * @private
+ */
+const _proxy = process.env.CORS_PROXY || '';
+
+/**
+ * Locale API endpoint
+ *
+ * @type {string}
+ * @private
+ */
+const _endpoint = `${_host}/common/v18/js/data/jsononly/locale`;
+
+/**
+ * Locale API class with method of fetching user's locale for
+ * ibm.com
+ */
+class LocaleAPI {
+  /**
+   * Gets the user's locale
+   *
+   * check if ipcinfo cookie exists (ipcinfoCookie util)
+   * if not, retrieve the user's locale through geolocation util + gets user's
+   * browser language preference then set the cookie
+   *
+   * @returns {object} object with lc and cc
+   * @example
+   * import { LocaleAPI } from '@carbon/ibmdotcom-services';
+   *
+   * async function getLocale() {
+   *   const locale = await LocaleAPI.getLocale();
+   *   return locale;
+   * }
+   */
+  static async getLocale() {
+    const cookie = ipcinfoCookie.get();
+    if (cookie && cookie.cc && cookie.lc) {
+      return cookie;
+    } else {
+      const cc = await geolocation();
+
+      /**
+       * get language preference from browser
+       * returns in en-US format so will need to extract language only
+       */
+      const lang = root.navigator.language;
+
+      const lc = lang.split('-')[0];
+
+      if (cc && lc) {
+        const list = await this.getList();
+        const verifiedCodes = this.verifyLocale(cc, lc, list);
+
+        // set the ipcInfo cookie
+        ipcinfoCookie.set(verifiedCodes);
+
+        return verifiedCodes;
+      }
+    }
+  }
+
+  /**
+   * Get the country list of all supported countries and their languages
+   *
+   * @returns {Promise <any>} promise object
+   *
+   * @example
+   * import { LocaleAPI } from '@carbon/ibmdotcom-services';
+   *
+   * function async getLocale() {
+   *    const list = await LocaleAPI.getList();
+   * }
+   */
+  static async getList() {
+    const url = `${_proxy}${_endpoint}/usen-locale.json`;
```

Comment:
@annawen1 shouldn't `getList` and `getLocale` not be hardcoded with `us` and `en`?

Reply:
hmm we only have the us-en country json file up, but also the list is used to verify the `cc-lc` so I don't want to use possibly non-existing `cc-lc` combo to make the call.

Reply:
since we have the redirect happening with `www.ibm.com/common/v18/js/data(([/\?].*)?$)` regex, I'm checking the response for now until we have a regex in place for the `locale` path

Reply:
This redirect has been approved, resolving